src/Kernel/Magic/Resolution/MagicResolver.cs

```diff
@@ -102,59 +102,76 @@ public MagicSymbol Resolve(string symbolName)
             return null;
         }
 
-        /// <summary>
-        /// Finds the MagicSymbols inside an assembly, and returns an instance of each.
-        /// </summary>
+        /// <inheritdoc />
         public IEnumerable<MagicSymbol> FindMagic(AssemblyInfo assm)
         {
             var result = new MagicSymbol[0];
 
-            if (cache.TryGetValue(assm, out result))
+            lock (cache)
```

Comment:
Before adding this `lock` statement, was this part not thread-safe?

Reply:
I think that's correct, yes. That said, it did not used to be called from background threads. This PR changes so that the engine starts searching for magic symbols in the background once started, rather than waiting for the first cell before searching.